### ISSUE TITLE
StreamExt is hard to find

### ISSUE THREAD

**Issue by jyn514** [rust-lang/futures-rs#2383]
This is more a user story than anything else. Say I have a `Stream` of futures and I want to do something with it. The first thing I did was go to https://docs.rs/futures/0.3.13/futures/, which links to https://docs.rs/futures/0.3.13/futures/stream/trait.Stream.html. Great, this looks like what I want! Then I look around on the page, and there's basically nothing there. `poll_next` looks complicated and low-level, and `size_hint` is not what I need. Then there's a very long list of "Implementors", which looks vaguely right, but I don't know which of them I need because they're not documented (presumably because I'm supposed to read the docs on `StreamExt`).

I figured this out by asking @Nemo157, but it would be nice to link directly there from the docs. I tried this originally, but it had no effect (probably because of https://github.com/rust-lang/rust/issues/81893):

```diff
diff --git a/futures-util/src/stream/mod.rs b/futures-util/src/stream/mod.rs
index f3b2baa4..c613843b 100644
--- a/futures-util/src/stream/mod.rs
+++ b/futures-util/src/stream/mod.rs
@@ -11,7 +11,11 @@
 
 #[cfg(feature = "alloc")]
 pub use futures_core::stream::{BoxStream, LocalBoxStream};
-pub use futures_core::stream::{FusedStream, Stream, TryStream};
+pub use futures_core::stream::FusedStream;
+/// Most functions on this trait are implemented through [`StreamExt`].
+pub use futures_core::stream::Stream;
+/// Most functions on this trait are implemented through [`TryStreamExt`].
+pub use futures_core::stream::TryStream;
 
 // Extension traits and combinators
 
```

**Comment by taiki-e** [comment]
thanks. first, we need to link to module instead of trait, in top page. (like io, channel, executor, etc.). next, we need to improve module docs (probably fine with copy of [std](https://doc.rust-lang.org/nightly/std/stream/index.html)).

**Comment by jyn514** [comment]
> first, we need to link to module instead of trait, in top page. (like io, channel, executor, etc.).

Sure, that could work.

> next, we need to improve module docs (probably fine with copy of std).

That doesn't help with StreamExt though - the std docs never mention any of the extension traits.

**Comment by taiki-e** [comment]
> That doesn't help with StreamExt though - the std docs never mention any of the extension traits.

you are right. copy is not enough, we need to add section or description for StreamExt and utils.
